src/csharp/ext/grpc_csharp_ext.c

```diff
@@ -927,7 +927,8 @@ GPR_EXPORT void GPR_CALLTYPE grpcsharp_metadata_credentials_notify_from_plugin(
 }
 
 typedef void(GPR_CALLTYPE *grpcsharp_metadata_interceptor_func)(
-  void *state, const char *service_url, grpc_credentials_plugin_metadata_cb cb,
+  void *state, const char *service_url, const char *method_name,
```

Comment:
Nice observation. I am doing this because of the way C# native interop works. If I passed the struct or pointer to the struct, I would need to redefine the same struct in C# code and make sure the memory layout of the C vs C# struct is the same. So for now, the approach I am using right now is less error-prone.